pkg/controller/proxycontroller.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-olm-operator/pkg/clients"
+)
+
+func NewProxyController(name string, proxyClient *clients.ProxyClient, operatorClient *clients.OperatorClient, eventRecorder events.Recorder) factory.Controller {
+	c := proxyController{
+		name:        name,
+		proxyClient: proxyClient,
+	}
+
+	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).WithInformers(proxyClient.Informer()).ToController(name, eventRecorder)
+}
+
+type proxyController struct {
+	name        string
+	proxyClient *clients.ProxyClient
+}
+
+func (c *proxyController) sync(ctx context.Context, _ factory.SyncContext) error {
+	logger := klog.FromContext(ctx).WithName(c.name)
+	logger.Info("sync started")
+	defer logger.Info("sync finished")
+
+	return UpdateProxyEnvironment(logger, c.proxyClient)
+}
+
+func UpdateProxyEnvironment(logger logr.Logger, pc *clients.ProxyClient) error {
+	logger.Info("getting proxy configuration")
+	proxySpec, err := pc.Get("cluster")
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info("proxy configuration not found")
+			updateEnvironment(logger, "HTTP_PROXY", "")
+			updateEnvironment(logger, "HTTPS_PROXY", "")
+			updateEnvironment(logger, "NO_PROXY", "")
+			return nil
+		}
+		return fmt.Errorf("failed to get proxy: %w", err)
+	}
+
+	logger.Info("updating environment")
+	updateEnvironment(logger, "HTTP_PROXY", proxySpec.Status.HTTPProxy)
```

Comment:
just a suggestion: maybe instead of taking the key/value as parameters, take a map? Then it might look like:

```
updateEnvironment(logger, map[string]string{
  "HTTP_PROXY"  : proxySpec.Status.HTTPProxy,
  "HTTPS_PROXY" : proxySpec.Status.HTTPSProxy,
  "NO_PROXY"    : proxySpec.Status.NO_PROXY,
})
```

Reply:
I can look, my concern would be in the function itself, as environments are not a simple `map`.

Reply:
Seems as though using maps makes the application of the environment variables occur in a non-deterministic order, which was not happening before. I may need to get the keys and sort them.

Reply:
I found a reference for this behavior in "Learning Go"!